src/transformers/modeling_tf_auto.py

```diff
@@ -140,126 +142,158 @@
 
 TF_MODEL_MAPPING = OrderedDict(
     [
+        (T5Config, TFT5Model),
```

Comment:
Reordered all the mappings here the same way it's done in PyTorch and added a test to check it's correct (also cc @Pierrci - think you reordered Roberta here recently).